tests/components/homematicip_cloud/test_init.py

```diff
@@ -1,154 +1,115 @@
     """Test that config for a accesspoint are loaded via config entry."""
-    with patch.object(hass, "config_entries") as mock_config_entries, patch.object(
-        hmipc, "configured_haps", return_value=[]
-    ):
-        assert (
-            await async_setup_component(
-                hass,
-                hmipc.DOMAIN,
-                {
-                    hmipc.DOMAIN: {
-                        hmipc.CONF_ACCESSPOINT: "ABC123",
-                        hmipc.CONF_AUTHTOKEN: "123",
-                        hmipc.CONF_NAME: "name",
-                    }
-                },
-            )
-            is True
-        )
 
-    # Flow started for the access point
-    assert len(mock_config_entries.flow.mock_calls) >= 2
+    entry_config = {
+        hmipc.CONF_ACCESSPOINT: "ABC123",
+        hmipc.CONF_AUTHTOKEN: "123",
+        hmipc.CONF_NAME: "name",
+    }
+    # no config_entry exists
+    assert len(hass.config_entries.async_entries(hmipc.DOMAIN)) == 0
+    # no acccesspoint exists
+    assert not hass.data.get(hmipc.DOMAIN)
+
+    assert (
+        await async_setup_component(hass, hmipc.DOMAIN, {hmipc.DOMAIN: entry_config})
+        is True
+    )
+
+    # config_entry created for access point
+    config_entries = hass.config_entries.async_entries(hmipc.DOMAIN)
+    assert len(config_entries) == 1
+    assert config_entries[0].data == {
+        "authtoken": "123",
+        "hapid": "ABC123",
+        "name": "name",
+    }
+    # defined access_point created for config_entry
+    assert isinstance(hass.data[hmipc.DOMAIN]["ABC123"], hmipc.HomematicipHAP)
 
 
 async def test_config_already_registered_not_passed_to_config_entry(hass):
     """Test that an already registered accesspoint does not get imported."""
-    with patch.object(hass, "config_entries") as mock_config_entries, patch.object(
-        hmipc, "configured_haps", return_value=["ABC123"]
-    ):
-        assert (
-            await async_setup_component(
-                hass,
-                hmipc.DOMAIN,
-                {
-                    hmipc.DOMAIN: {
-                        hmipc.CONF_ACCESSPOINT: "ABC123",
-                        hmipc.CONF_AUTHTOKEN: "123",
-                        hmipc.CONF_NAME: "name",
-                    }
-                },
-            )
-            is True
-        )
-
-    # No flow started
-    assert not mock_config_entries.flow.mock_calls
 
+    mock_config = {
+        hmipc.HMIPC_AUTHTOKEN: "123",
+        hmipc.HMIPC_HAPID: "ABC123",
+        hmipc.HMIPC_NAME: "name",
+    }
+    MockConfigEntry(domain=hmipc.DOMAIN, data=mock_config).add_to_hass(hass)
+
+    # one config_entry exists
+    config_entries = hass.config_entries.async_entries(hmipc.DOMAIN)
+    assert len(config_entries) == 1
+    assert config_entries[0].data == {
+        "authtoken": "123",
+        "hapid": "ABC123",
+        "name": "name",
+    }
+    # config_enty has no unique_id
+    assert not config_entries[0].unique_id
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4887152)